server/controllers/auth.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable consistent-return */
+import jwt from 'jsonwebtoken';
+
+import config from '../config';
+import User from '../models/User';
+
+function invalidResponse(response) {
+  return response.status(400).json({
+    success: false,
+    message: 'Invalid Email/Password.',
+    user: null,
+  });
+}
+
+function getToken(userData) {
+  return jwt.sign(userData, config.JWT_SECRET_KEY, { expiresIn: '7d' });
+}
+
+function successfulResponse(response, message, userData, token) {
+  return response.status(200).json({
+    success: true,
+    message,
+    user: { ...userData, token },
+  });
+}
+
+function tokenErrorResponse(response) {
+  return response.status(400).json({
+    success: false,
+    message: 'Token could not be created',
+    user: null,
```

Comment:
@avinash360 I think `user: null` is not required when `success: false`. I think we need to change the structure of our error object